geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -259,24 +272,24 @@ object DeltaWriter extends StrictLogging {
     * @param threadedBatches record batches, grouped by threading key
     * @return
     */
-  private def reduceNoSort(sft: SimpleFeatureType,
-                           dictionaryFields: Seq[String],
-                           encoding: SimpleFeatureEncoding,
-                           mergedDictionaries: MergedDictionaries,
-                           batchSize: Int,
-                           threadedBatches: Array[Array[Array[Byte]]])
-                          (implicit allocator: BufferAllocator): CloseableIterator[Array[Byte]] = {
```

Comment:
Ok, why this change?  (This is fairly deep in the code, and I'll admit to being lost.:))

Reply:
instead of passing in an allocator, a new one is created and wrapped into the lifecycle of the close() method on the result iterator.